### PR TITLE
Add pipenv

### PR DESCRIPTION
**Pull request by Seth10001**
I added the python package manager tool called `pipenv`.  It allows for a smoother transition when adding new modules.

However, all existing users will need to upgrade to `pipenv`. Otherwise, the make command will fail. It might be possible to modify the make files to check if the commands already exist and use that instead.

Thoughts? @John-Carr @scuervo101 @mmhobi7

### CODE REVIEW

**Comment by mmhobi7** [comment]
Why pipx? We could jump right over to just pipenv with python -m pip install pipenv. Also the minimum required python version is set (in the pipfile) to 3.9 it can be lowered to 3.8. Also the readme states the python3.7 is needed. 

Besides that looks good!

**Comment by Seth10001** [comment]
@mmhobi Oh hm, I was just following a guide. I could change it to download `pipenv` from `pip` directly. Good point on the python version!

**Review by scuervo101 — APPROVED** [review]
Like Muhamed said, I got pipenv working by just `pip install`ing it. Everything looks good, just update the version number in the pipfile as it's at 3.9 currently.

**Comment by Seth10001** [comment]
Sadly there wasn't a way to allow python versions 3.7 and above in the pipfile. I was, however, able to allow ANY python 3 version. Also, it appears that read the docs, requires there to be a requirements.txt file, so I added some instructions on how to generate that from pipenv.

**Comment by Seth10001** [comment]
Oh and I added a few legacy commands in case some people can't get pipenv working.

`make legacy-livehtml`
`make legacy ${COMMAND}`

File: requirements.txt [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
-sphinxcontrib-devhelp==1.0.2
-sphinxcontrib-htmlhelp==1.0.3
-sphinxcontrib-httpdomain==1.7.0
-sphinxcontrib-jsmath==1.0.1
-sphinxcontrib-qthelp==1.0.3
-sphinxcontrib-serializinghtml==1.1.4
-tornado==6.0.4
-urllib3==1.25.9
-wincertstore==0.2
+-i https://pypi.org/simple
```
Comment (mmhobi7) [inline requirements.txt]: mistake?

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+[packages]
+sphinx-autobuild = "2020.9.1"
+sphinx-rtd-theme = "0.5.0"
+recommonmark = "0.6.0"
+"sphinxcontrib.httpdomain" = "1.7.0"
+sphinx-tabs = "1.3.0"
+
+[requires]
+python_version = "3"
```
Comment (mmhobi7) [inline Pipfile:16]: Since sphinx requires a minimum of 3.5, just edit this to be "3.5", also update the readme

**Comment by mmhobi7** [comment]
Looks good

File: Pipfile (hunk 2 above)
Comment (Seth10001) [reply Pipfile:16]: Sadly, pipenv doesn't support minimum python versions. If we make it 3.5 then any python version above that will give us an error :(

**Review by John-Carr — APPROVED** [review]
Looks good to me.

File: Pipfile (hunk 2 above)
Comment (mmhobi7) [reply Pipfile:16]: Yeah you're right my b

**Pull request merged**